src/reanimated2/layoutReanimation/sharedTransitions/index.ts

```diff
@@ -1,40 +1,90 @@
 import { withTiming } from '../../animation';
 import {
-  ILayoutAnimationBuilder,
-  LayoutAnimationFunction,
-  LayoutAnimationsValues,
+  SharedTransitionAnimationsFunction,
+  SharedTransitionAnimationsValues,
+  ProgressAnimationCallback,
+  LayoutAnimationType,
 } from '../animationBuilder/commonTypes';
 import { StyleProps } from '../../commonTypes';
+import { configureLayoutAnimations } from '../../core';
+import { Platform } from 'react-native';
 
 const supportedProps = ['width', 'height', 'originX', 'originY', 'transform'];
 
-type AnimationFactoryType = (values: LayoutAnimationsValues) => StyleProps;
+type AnimationFactory = (
+  values: SharedTransitionAnimationsValues
+) => StyleProps;
 
-export class SharedTransition implements ILayoutAnimationBuilder {
-  animationFactory: AnimationFactoryType | null = null;
-  defaultDuration = 500;
+export class SharedElementTransition {
+  private _animationFactory: AnimationFactory | null = null;
+  private _animation: SharedTransitionAnimationsFunction | null = null;
+  private _transitionDuration = 500;
+  private _customProgressAnimation?: ProgressAnimationCallback = undefined;
+  private _progressAnimation?: ProgressAnimationCallback = undefined;
 
-  static createInstance(): SharedTransition {
-    return new SharedTransition();
+  public animation(
+    animationFactory: AnimationFactory
+  ): SharedElementTransition {
+    this._animationFactory = animationFactory;
+    return this;
   }
 
-  static custom(animationFactory: AnimationFactoryType): SharedTransition {
-    return this.createInstance().custom(animationFactory);
+  public progressAnimation(
+    progressAnimationCallback: ProgressAnimationCallback
+  ): SharedElementTransition {
+    this._customProgressAnimation = progressAnimationCallback;
+    return this;
   }
 
-  custom(animationFactory: AnimationFactoryType): SharedTransition {
-    this.animationFactory = animationFactory;
+  public duration(duration: number): SharedElementTransition {
+    this._transitionDuration = duration;
     return this;
   }
 
-  static build(): LayoutAnimationFunction {
-    return this.createInstance().build();
+  public registerTransition(
+    viewTag: number,
+    sharedTransitionTag: string
+  ): void {
+    const transitionAnimation = this.getTransitionAnimation();
+    const progressAnimation = this.getProgressAnimation();
+    let progressAnimationPriority = 'primary';
+    if (this._animationFactory) {
+      if (Platform.OS === 'ios') {
+        progressAnimationPriority = 'secondary';
```

Comment:
Instead of keeping these as strings, can we convert them to numeric enums or just use a boolean flag?